tests/test___init__.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the functions contained in pennylane/__init__.py.
+"""
+# pylint: disable=protected-access,cell-var-from-loop
+
+import pytest
+import pennylane as qml
+
+
+class TestPennyLaneInit:
+    """Unit tests for the functions in the pennylane/__init__.py."""
+
+    @pytest.mark.usefixtures("skip_if_no_pennylane_qiskit_support")
+    def test_plugin_converters(self):
+        """Test that the converter from the PennyLane-Qiskit plugin entry points are included
+        in the correct entry points."""
+        assert 'qiskit' in qml.plugin_converters
+        assert 'qasm' in qml.plugin_converters
+        assert 'qasm_file' in qml.plugin_converters
+
+    @pytest.mark.usefixtures("skip_if_no_pennylane_qiskit_support")
+    def test_load(self):
+        """Test that a converter from the PennyLane-Qiskit plugin is returned."""
+
+        from qiskit import QuantumCircuit
+
+        qc = QuantumCircuit(2)
+        assert callable(qml.load(qc, name='qiskit'))
```

Comment:
I'm not sure the above two tests should be in the PennyLane test suite. Instead, since they rely on an additional plugin, they should be integration tests in the PennyLane-Qiskit repo (since they are testing logic in the PennyLane-Qiskit package). Similar to how external device entry points aren't tested in PennyLane.

This also avoids circular travis dependencies, where the PL-Core travis build requires the latest github master of PL-Qiskit, and the PL-Qiskit travis build requires the latest github master of PL-Core

Reply:
I agree, that integration tests could be included in PennyLane-Qiskit to test the entire logic behind how ``load`` functions.

However, I believe that it is also worth checking whatever small purpose ``__init__.py/load`` has in PennyLane core. It is actually not closely related to how the logic is defined for each of the ``load`` functions in the plugins. What this one does is simply "fetch" the available ``load`` function or raise an error. I'm open to suggestions what else could be checked here, but I feel it would be important to be able to distinguish whether there would possibly be an error with how these ``load`` functions are fetched and with the specific ``load`` functions themselves (which we can assume here are checked in the related plugins).

Could such circular travis dependencies arise when the imports are put under condition?

Reply:
I'm happy with the logic of `load` in PL-Core being tested, if it doesn't require an external plugin be installed - perhaps there is a 'minimal' integration test we could perform? Or perhaps a way of `mocking out` a plugin with entry points?

Reply:
It is a good point not to use an external plugin, I agree!

So far I have found how to add a fake entrypoint and get it discovered in a test, however, I still have yet to add a mock module with a function.

As I feel this is not at all priority, I'm removing these tests from the PR so that the main functionality can be merged.

Reply:
Sounds good. Perhaps we should make an issue to remind ourselves to add tests for these in the future.